### PR TITLE
Update interop tests to use modern package lock file

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

File: src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs [hunk 1]
```diff
@@ -80,33 +80,35 @@ protected override string PeerCore
...
-                        _peer = "unknown"; // Match Grpc.Core
-                    }
+                    _peer = BuildPeer();
                 }
 
                 return _peer;
             }
         }
 
+        private string BuildPeer()
```
Comment (JamesNK) [inline src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs:90]: Mirror update in GrpcHttpApi. Checked in by accident but IMO not worth a separate PR

File: testassets/InteropTestsGrpcWebWebsite/Tests/package.json [hunk 2]
```diff
@@ -6,8 +6,8 @@
     "test": "jest test"
   },
   "dependencies": {
-    "puppeteer": "^11.0.0",
-    "jest": "^27.3.1",
-    "jest-puppeteer": "^6.0.0"
+    "puppeteer": "^13.3.2",
```
Comment (captainsafia) [inline testassets/InteropTestsGrpcWebWebsite/Tests/package.json:9]: Not super relevant but do we want to pin these to the patch version for more consistency in the build?

Not sure if this repo is set up with dependabot which would defeat the purpose but just a thought

Comment (JamesNK) [reply testassets/InteropTestsGrpcWebWebsite/Tests/package.json:9]: Doesn't the lock file pin dependencies for the build?

I'm not a node expert but the ^ syntax is used in package.json files in aspnetcore repo.

File: testassets/InteropTestsGrpcWebWebsite/Tests/custom-environment.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
-// custom-environment.js
+// For some reason the expect-puppeteer require is returning a module rather than the function.
+// Helper function unwraps the module and returns the inner function.
+function _interopDefault(ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
+
 const PuppeteerEnvironment = require('jest-environment-puppeteer');
-const expect = require('expect-puppeteer');
+const expect = _interopDefault(require('expect-puppeteer'));
```
Comment (captainsafia) [inline testassets/InteropTestsGrpcWebWebsite/Tests/custom-environment.js:6]:
```suggestion
import expect from 'expect-puppeteer'
```

Can we use module imports here instead?

Comment (JamesNK) [reply testassets/InteropTestsGrpcWebWebsite/Tests/custom-environment.js:6]: I tried that but got an error along the lines of "can't import outside of a module"

Comment (JamesNK) [reply testassets/InteropTestsGrpcWebWebsite/Tests/custom-environment.js:6]: Hmm, I didn't do exactly what you suggested: [`554f1bc` (#1615)](https://github.com/grpc/grpc-dotnet/pull/1615/commits/554f1bc48d494e95c8c42f10a2cdc28fca03453c)

I'll give it a try

Comment (JamesNK) [reply testassets/InteropTestsGrpcWebWebsite/Tests/custom-environment.js:6]: To get import to work the package type has to be module which means lots of other changes. The current fix probably isn't better but it is simpler.

File: testassets/InteropTestsGrpcWebWebsite/Tests/package.json (hunk 2 above)
Comment (captainsafia) [reply testassets/InteropTestsGrpcWebWebsite/Tests/package.json:9]: Yes, it does! Missed that there were lock files in here. 😅

**Review by captainsafia — APPROVED** [review]

**Pull request merged**